src/test/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/service/DecrypterTest.java

```diff
@@ -0,0 +1,275 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.kafka.KafkaAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.function.context.config.ContextFunctionCatalogAutoConfiguration;
+import org.springframework.cloud.stream.binder.Binder;
+import org.springframework.cloud.stream.binder.BinderConfiguration;
+import org.springframework.cloud.stream.config.BinderFactoryAutoConfiguration;
+import org.springframework.cloud.stream.config.BindingServiceConfiguration;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.function.FunctionConfiguration;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.cloud.stream.messaging.DirectWithAttributesChannel;
+import org.springframework.context.ApplicationContext;
+import org.springframework.kafka.annotation.KafkaListenerAnnotationBeanPostProcessor;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.support.ProducerListener;
+import org.springframework.messaging.Message;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.BlobApplicationAware;
+import it.gov.pagopa.rtd.ms.rtdmsdecrypter.model.EventGridEvent;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.kafka.clients.NetworkClient;
+import org.bouncycastle.bcpg.ArmoredOutputStream;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPCompressedData;
+import org.bouncycastle.openpgp.PGPCompressedDataGenerator;
+import org.bouncycastle.openpgp.PGPEncryptedData;
+import org.bouncycastle.openpgp.PGPEncryptedDataGenerator;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPLiteralData;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcePGPDataEncryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyKeyEncryptionMethodGenerator;
+
+@SpringBootTest
+@ContextConfiguration(classes = {Decrypter.class})
+@TestPropertySource(value = {"classpath:application-nokafka.yml"}, inheritProperties = false)
+class DecrypterTest {
+
+
+  @Autowired
+  Decrypter decrypter;
+
+  @Autowired
+  ApplicationContext context;
+
+  // @MockBean
```

Comment:
very minor issue but still: there is some value in keeping all this commented code here?

Reply:
No. It's superseded by the use of a custom property file. Also the `context` no longer needs to be auto-wired

Reply:
> @GiovanniMancini do you think could be beneficial to add an endpoint to manually trigger the decryption of a file (not in this PR)? I think that in case of some failure it could be cumbersome to re-trigger the decryption by injecting the correct message in the queue, but maybe I'm overestimating it: what do you think?

Thank you for the idea, but in my opinion having and endpoint doesn't increase controllability of the service. We already have the external endpoint to submit encrypted files, we could re-trigger a new decryption by submitting a new file. An endpoint exposed by the micro-service would allow to test event delivery in isolation, but this is a platform feature, I think it's better tested end to end